finals-2024-01-15/code/5265-genci-telqiu/src/app/movie-list/movie-list.component.ts

```diff
@@ -1,10 +1,70 @@
-import { Component } from '@angular/core';
+import { Component, OnInit } from '@angular/core';
+import { MovieService } from '../services/movie.service';
 
 @Component({
   selector: 'app-movie-list',
   templateUrl: './movie-list.component.html',
-  styleUrl: './movie-list.component.css'
+  styleUrls: ['./movie-list.component.css'],
 })
-export class MovieListComponent {
+export class MovieListComponent implements OnInit {
+  movies: any[] = [];
+  filteredMovies: any[] = [];
+  genres: string[] = [];
+  titleFilter: string = '';
+  yearFilter: number | null = null;
+  selectedGenre: string | null = null;
+  ratingFilter: number | null = null;
+
+  constructor(private movieService: MovieService) {}
+
+  ngOnInit(): void {
+    this.movieService.getMovies().subscribe((data) => {
+      this.movies = data;
+      this.filteredMovies = this.movies;
+      this.extractGenres();
+    });
+  }
+
+  private extractGenres(): void {
```

Comment:
ok code, but the genres were supposed to be gotten from the backend 😄  + 5 points